File: src/main/java/com/shelterhub/domain/model/Animal.java
package com.shelterhub.domain.model;

import com.shelterhub.domain.enums.AnimalType;
import com.shelterhub.domain.enums.Gender;
import com.shelterhub.domain.enums.Size;
import com.shelterhub.dto.response.AnimalResponse;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table
public class Animal {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    @Column(nullable = false)
    private String name;
    private String identification;
    private String behavior;
    private String breed;
    private String history;
    private LocalDate estimatedAge;
    private UUID medicalRecordId;
    private AnimalType animalType;
    private Gender gender;
    private Size size;

    public AnimalResponse toResponse() {
        return AnimalResponse.builder()
                .id(getId())
                .name(getName())
                .identification(getIdentification())
                .behavior(getBehavior())
                .breed(getBreed())
                .history(getHistory())
                .estimatedAge(getEstimatedAge())
                .medicalRecordId(getMedicalRecordId())
                .animalType(String.valueOf(getAnimalType()))
                .gender(getGender())
                .size(getSize())
                .build();
    }
}

File: src/main/kotlin/br/com/shelterhubmanagementapi/domain/model/Animal.kt
package br.com.shelterhubmanagementapi.domain.model

import br.com.shelterhubmanagementapi.domain.enums.AnimalType
import br.com.shelterhubmanagementapi.domain.enums.Gender
import br.com.shelterhubmanagementapi.domain.enums.Size
import br.com.shelterhubmanagementapi.dto.response.AnimalResponse
import kotlinx.datetime.LocalDate
import java.util.UUID

data class Animal(
    val id: UUID = UUID.randomUUID(),
    val name: String,
    val identification: String,
    val behavior: String,
    val breed: String,
    val history: String,
    val estimatedAge: LocalDate? = null,
    val medicalRecordId: UUID,
    val animalType: AnimalType,
    val gender: Gender,
    val size: Size,
)

fun Animal.toResponse(): AnimalResponse =
    AnimalResponse(
        id = this.id,
        name = this.name,
        identification = this.identification,
        behavior = this.behavior,
        breed = this.breed,
        history = this.history,
        estimatedAge = this.estimatedAge.toString(),
        medicalRecordId = this.medicalRecordId,
        animalType = this.animalType.toString(),
        gender = this.gender,
        size = this.size,
    )
